armi/operators/operator.py

```diff
@@ -618,6 +633,58 @@ def interactAllCoupled(self, coupledIteration):
         activeInterfaces = [ii for ii in self.interfaces if ii.enabled()]
         self._interactAll("Coupled", activeInterfaces, coupledIteration)
 
+    def _tightCouplingOldValues(self):
+        activeInterfaces = [ii for ii in self.interfaces if ii.enabled()]
+        for interface in activeInterfaces:
+            interface.tightCouplingOldValue = interface.getTightCouplingValue()
+
+    def _computeTightCouplingConvergence(self):
```

Comment:
Just spit-balling.

We are adding a lot of lines of code to the primary `Operator` class here.

Would it be more clean to move these methods into their own little class? `_computeTightCouplingConvergence`, `_printTightCouplingReport`, `_checkTightCouplingConvergence`, `_printTightCouplingReport`?

It would take the interfaces in its constructor and would keep the `self.convergenceSummary` as an attribute.

**Pros**:

* It might be cleaner
* It might make it easier for future readers to understand the code.

**Cons**:

* Not necessary
* It's an awfully small class.

Maybe.  I'm obviously on the fence though.  Food for thought.

Reply:
Hm @john-science - want to talk about this? I want to help @albeanth get this is as soon as possible. I am not sure if we want to add another class/object to this, but it might not be a bad optimization/improvement for the future?

Reply:
We should talk.  

The `Operator` is the most important class in ARMI.  It controls how `Plugin`s and `Interface`s become simulations.  It does not have ANY physics or science code in it.

Let's look at the methods in this class:

```python
    def burnSteps(self):
    def maxBurnSteps(self):
    def stepLengths(self):
    def cycleLengths(self):
    def availabilityFactors(self):
    def cycleNames(self):
    def atEOL(self):
    def initializeInterfaces(self, r):
    def operate(self):
    def printInterfaceSummary(self, interface, interactionName, statePointIndex, *args):
    def interactAllInit(self):
    def interactAllBOL(self, excludedInterfaceNames=()):
    def interactAllBOC(self, cycle):
    def interactAllEveryNode(self, cycle, tn, excludedInterfaceNames=None):
    def interactAllEOC(self, cycle, excludedInterfaceNames=None):
    def interactAllEOL(self):
    def interactAllCoupled(self, coupledIteration):
    def interactAllError(self):
    def createInterfaces(self):
    def addInterface():
    def removeAllInterfaces(self):
    def removeInterface(self, interface=None, interfaceName=None):
    def getInterface(self, name=None, function=None):
    def interfaceIsActive(self, name):
    def getInterfaces(self):
    def reattach(self, r, cs=None):
    def detach(self):
    def loadState():
    def snapshotRequest(self, cycle, node):
    def setStateToDefault(cs):
    def couplingIsActive(self):
```

See how this are all about code cruft things?  Like "init an Interface", "run an interface", "add an interface", etc...

There is NO physics in this file.  None.  After a solid decade of work, no one on the ARMI team has every found it necessary to inject a bunch of physics code into this file/class.  I strongly suggest we not start mixing science code into this file now.

Reply:
Let's chat offline this afternoon to brainstorm and workshop this.

Long story short, I do not intend to do any physics with this new code. Rather, we are simply checking to see if active interfaces have numerically converged by quantitatively measuring values through various norms. So we aren't really doing physics, but rather a touch of math and book keeping.

Reply:
@ntouran I would love your opinion on these changes to the `Operator` class.

Reply:
@john-science I believe this comment has been addressed with the interfaces.py::TightCoupler class. Let me know what you think?

Reply:
I think its addressed. resolving.